lib/rucio/tests/common.py

```diff
@@ -66,24 +67,36 @@ def scope_name_generator():
     return 'mock_' + str(uuid()).lower()[:16]
 
 
-def did_name_generator(did_type='file'):
+def did_name_generator(did_type='file', name_prefix='', name_suffix='', cnt=0, path=None):
     """ Generate random did name.
+    :param did_type: A string to create a meaningful did_name depending on the did_type (file, dataset, container)
+    :param prefix: String to prefix to the did name
+    :param suffix: String to append to the did name
+    :param cnt: A counter to include in the did_name
+    :param path: If specified, use the path to generate the did_name
 
     :returns: A random did name
     """
     if os.getenv('POLICY') == 'belleii':
-        path = '/belle'
-        path += '/cont_%s' % str(uuid())
+        if path is not None:
+            return path
+        path = '/belle/mock' + name_prefix
+        if path[-1] != '/':
+            path += '/'
+        path += 'cont_%s' % str(uuid())
         if did_type == 'container':
             return path
-        path += '/dataset_%s' % str(uuid())
+        path += 'dataset_%s' % str(uuid())
         if did_type == 'dataset':
             return path
-        path += '/file_%s' % str(uuid())
+        path += 'file_%s' % str(uuid())
         return path
```

Comment:
Handling paths by string manipulation is error-prone and hard to read, how about using the `os.path` module?

This should translate to something like:
```Python
if path is not None:
    return path

container_path = os.path.join("/belle", "mock", name_prefix, 'cont_%s' % str(uuid()))
if did_type == 'container':
    return container_path

dataset_path = os.path.join(container_path, 'dataset_%s' % str(uuid()))
if did_type == 'dataset':
    return dataset_path

file_path = os.path.join(dataset_path, 'dataset_%s' % str(uuid()))
return file_path
```

Or something like that?

Reply:
Good suggestion. Implemented.